server/sonar-server/src/main/java/org/sonar/server/rule/RuleService.java

```diff
@@ -56,20 +55,19 @@ public RuleService(RuleIndex index, RuleUpdater ruleUpdater, RuleCreator ruleCre
   }
 
   @CheckForNull
+  @Deprecated
   public Rule getByKey(RuleKey key) {
-    return index.getNullableByKey(key);
+    throw new UnsupportedOperationException("Please use RuleDao");
   }
 
+  @Deprecated
   public List<Rule> getByKeys(Collection<RuleKey> keys) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)